edge/yaml/yaml.go

```diff
@@ -0,0 +1,143 @@
+package yaml
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	v1Types "k8s.io/api/core/v1"
+	v1AMacTypes "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+
+	"github.com/pkg/errors"
+	"github.com/portainer/agent"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+type yaml struct {
```

Comment:
this file is basically adding `imagePullSecrets` so the name `yaml` could be confusing?